doc/sphinx/how-to/how-to-update-symbols-map.md

```diff
@@ -0,0 +1,217 @@
+# How to Update Symbols Map Files
+The Mir project is a collection of C++ libraries that a consumer
+can use to create a Wayland compositor. In order for consumers
+of Mir's libraries to be confident in what they're building against, the
+symbols of each library are versioned.
+
+To describe these versions, each user-facing library defines a `symbols.map`
+file. These files are comprised of version stanzas which contain the symbols
+associated with that version. There are `symbols.map` files for the following
+libraries, although only the first three are typically used in practice:
+
+- [miral](#how-to-update-miral-symbols)
+- [miroil](#how-to-update-miroil-symbols)
+- [mirserver](#how-to-update-mirserver-symbols-map)
+- mircore
+- mircommon
+- mirplatform
+- mirwayland
+
+It is tedious to edit these files. Luckily for us, this versioning business
+is (mostly) automated so that we have to think very little about it.
+
+## When are symbols.map files updated
+Two different circumstances will prompt a developer to update
+these files:
+
+1. **A new symbols is added to a library**
+
+    This means that we have extended the existing interface.
+    The new symbols can simply be put in a new stanza with a "bumped"
+    minor version. Note that the minor version must only be bumped once
+    per release cycle. This means that if I add a new symbol two months
+    before a release and you add a symbol 1 day before the release, then
+    our new symbols will go in the same stanza. In that situation, I
+    would have been the one to create the new stanza.
+
+2. **A symbol is changed or removed** (e.g. a new parameter is
+   added to a method call, a class is removed, etc.)
+
+    This is an API break. This means that a consumer of the library
+    can no longer safely compile against the new version of that library
+    without maybe breaking their build. In this scenario, we are forced
+    to create a single new stanza at a brand new version in the `symbols.map`
+    file. This new stanza will contain all of the symbols in that library.
+
+Sometimes we break a symbols but don't even realize it until someone
+points it out in a pull request. Nowadays, the symbols check is automated
+by CI. You can find results of this check
+[here](https://github.com/canonical/mir/actions/workflows/symbols-check.yml).
+This check is run on each pull request. It will inform you whether or not
+a `symbols.map` file needs to be updated due to your change.
+
+**Warning**: This check is not sophisticated to know if you change the
```

Comment:
I can't quite parse what you're trying to say here

Reply:
Updated! let me know if it reads better to you now

Reply:
Crystal clear now :+1: